### PR TITLE
Added partial support for node:os > userInfo function

### PR DESCRIPTION
**Pull request by zeachco**
Most packages in node uses os.userInfo to read username, shell or homedir. this changes still causes an exception on trying to read `uid` and `gid` and supports only `utf-8` option for now but that's a start.

# impact
While this isn't an accurate implementation, some npm packages such as https://www.npmjs.com/package/atomically (which is used by remix and many other projects) depends on it and would make a large chunk of the npm ecosystem workable, making deno more adoptable

this is linked to https://github.com/denoland/deno/issues/17850

This feels like a "poorman solution" but if any insights on integrating [gu]id is provided I would gladly try my best to address it.

propositions:
- maybe we could put that behind an unstable flag while it's a temp / partial implementation?
something like `deno run --unstable-os-user-info npm:<package>`

image

### CODE REVIEW

File: ext/node/polyfills/os.ts [hunk 1]
```diff
@@ -315,12 +316,28 @@ export function uptime(): number {
...
   options: UserInfoOptions = { encoding: "utf-8" },
 ): UserInfo {
-  notImplemented(SEE_GITHUB_ISSUE);
+  if (options.encoding !== "utf-8") notImplemented(SEE_GITHUB_ISSUE);
+
+  const username = Deno.env.get("USER");
+  const homedir = Deno.env.get("HOME");
+  const shell = Deno.env.get("SHELL") || null;
+
+  if (!username || !homedir) throw codes.SYSERR;
```
Comment (zeachco) [inline ext/node/polyfills/os.ts:332]: should I throw `notImplemented(SEE_GITHUB_ISSUE);` here too? node does a system error when those aren't set

File: ext/node/polyfills/os.ts [hunk 2]
```diff
@@ -315,12 +316,28 @@ export function uptime(): number {
...
+
+  return {
+    uid: -1,
+    gid: -1,
```
Comment (zeachco) [inline ext/node/polyfills/os.ts:336]: should we warn about uid/gid in the cli?

**Comment by bartlomieju** [comment]
Thanks for the PR @zeachco, FYI there's already a PR that implements these APIs (https://github.com/denoland/deno/pull/19370) and it's supposed to land before v1.36 is released.

**Comment by zeachco** [comment]
Awesome, I wasn't sure if that or would cut in the next version hence the quick minimalistic one. Great to hear. I'll close this one

**Pull request closed without merging**